File: C++/59. Spiral Matrix II.cpp
class Solution {
public:
    vector<vector<int>> generateMatrix(int n) {
        vector<vector<int> >ans(n, vector<int>(n));
        int rs = 0, re = n-1;
        int cs = 0, ce = n-1;
        int cnt = 1;
        while(rs <= re && cs <= ce)
        {
            for(int i=cs;i<=ce;i++)
                ans[rs][i] = cnt++;
            rs++;
            for(int i=rs;i<=re;i++)
                ans[i][ce] = cnt++;
            ce--;
            for(int i=ce;i>=cs;i--)
                ans[re][i] = cnt++;
            re--;
            for(int i=re;i>=rs;i--)
                ans[i][cs] = cnt++;
            cs++;
        }
        return ans;
    }
};

File: Python/59. Spiral Matrix II.py
class Solution:
    def generateMatrix(self, n):
        """
        :type n: int
        :rtype: List[List[int]]
        """
        ans = [[0 for i in range(n)] for j in range(n)]
        cnt = 1
        rs, re = 0, n - 1
        cs, ce = 0, n - 1
        while rs <= re and cs <= ce:
            for i in range(cs, ce+1):
                ans[rs][i] = cnt
                cnt += 1
            rs += 1
            for i in range(rs, re+1):
                ans[i][ce] = cnt
                cnt += 1
            ce -= 1
            for i in range(ce, cs-1, -1):
                ans[re][i] = cnt
                cnt += 1
            re -= 1
            for i in range(re, rs-1, -1):
                ans[i][cs] = cnt
                cnt += 1
            cs += 1
        return ans